File: 1.two-sum.py
# @leet start
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        hash = {}
        for i in range(0,len(nums)):
            num=nums[i]
            if target - num in hash:
                return [i, hash[target - num]]
            hash[num]=i
        return []


# @leet end

File: 1.two-sum.ts
// @leet start
function twoSum(nums: number[], target: number): number[] {
	const map: Map<number, number> = new Map();
	for (let i = 0; i < nums.length; i++) {
		const num = nums[i];
		if (map.has(target - num)) {
			return [map.get(target-num)!, i]
		}
		map.set(num, i)
	}
	return []
}
// @leet end
